### PR TITLE
feat: generate machine secret

### PR DESCRIPTION
**Pull request by stebenz**
### Definition of Ready

- [ ] Short description of the feature/issue is added in the pr description
- [ ] PR is linked to the corresponding user story
- [ ] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [ ] No debug or dead code
- [ ] My code has no repetitions
- [ ] All non-functional requirements are met
- [ ] The generic lifecycle acceptance test passes for affected resources.
- [ ] Examples are up-to-date and meaningful. The provider version is incremented.
- [ ] Docs are generated.
- [ ] Code is generated where possible.

### CODE REVIEW

File: zitadel/machine_user/resource.go [hunk 1]
```diff
@@ -60,6 +60,24 @@ func GetResource() *schema.Resource {
...
+			clientIDVar: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Value of the client ID",
+				Sensitive:   true,
+			},
+			clientSecretVar: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Value of the client secret",
```
Comment (eliobischof) [inline zitadel/machine_user/resource.go:78]:
```suggestion
				Description: "Value of the client secret if withSecret is true",
```

**Review by eliobischof — CHANGES_REQUESTED** [review]

File: zitadel/machine_user/resource.go [hunk 2]
```diff
@@ -60,6 +60,24 @@ func GetResource() *schema.Resource {
 				},
 				Default: defaultAccessTokenType,
 			},
+			generateSecretVar: {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Generate machine secret, only applicable if creation or change from false",
+			},
```
[the remark this answers is not in the document]
Comment (stebenz) [reply zitadel/machine_user/resource.go:68]: Sounds good to me

**Review by eliobischof — APPROVED** [review]

**Comment by swaroopar** [comment]
@stebenz Thank you for this.  It is exactly what we are looking for.

**Review by eliobischof — CHANGES_REQUESTED** [review]
Please have a look at this https://github.com/zitadel/terraform-provider-zitadel/pull/162

File: docs/index.md [hunk 3]
```diff
@@ -31,7 +31,7 @@ terraform {
   required_providers {
     zitadel = {
       source  = "zitadel/zitadel"
-      version = "1.0.0-alpha.16"
+      version = "1.0.7"
```
Comment (eliobischof) [inline docs/index.md:34]: It's a feature. Plz also change the PR title.

**Review by eliobischof — CHANGES_REQUESTED** [review]

File: docs/index.md (hunk 3 above)
Comment (stebenz) [reply docs/index.md]: changed to feat and new version after release

**Pull request merged**